### PR TITLE
Add changes to NewChannel page

### PR DESCRIPTION
**Pull request by usmanabdulmajid**
I'm making this PR so my teammate can make the api implementation

https://user-images.githubusercontent.com/63570223/133859127-b48c626b-ae8f-46b4-9dcb-7cebc1c86166.mp4

on

### CODE REVIEW

File: lib/ui/view/channel/new_channel/new_channel.dart [hunk 1]
```diff
@@ -57,54 +70,202 @@ class _NewChannelState extends State<NewChannel> {
...
+                              border: InputBorder.none,
+                              hintText: 'e.g team-uchiha',
+                              hintStyle: TextStyle(
+                                fontWeight: FontWeight.w400,
+                                color: Color(0xFFA1A9B2),
+                              ),
+                            ),
+
+                            onChanged: (val) {
+                              setState(() {
```
Comment (isaacadariku) [inline lib/ui/view/channel/new_channel/new_channel.dart:113]: Move the logic to the viewmodel and get the values needed over there.

File: lib/ui/view/channel/new_channel/new_channel.dart [hunk 2]
```diff
@@ -57,54 +70,202 @@ class _NewChannelState extends State<NewChannel> {
...
-                                style: AppTextStyles.body3Medium,
-                                children: <TextSpan>[
-                                  TextSpan(
-                                    text: ' e.g team Uhciha',
-                                    style: AppTextStyles.body2Medium,
-                                  ),
-                                ]),
+                    padding: const EdgeInsets.only(left: 16.0),
+                    child: Visibility(
+                      visible: _hasError,
```
Comment (isaacadariku) [inline lib/ui/view/channel/new_channel/new_channel.dart:133]: _hasError is suppose to be a viewmodel logic, so we can test all this.

move the logic there, ensure your UI have no logic

File: lib/ui/view/channel/new_channel/new_channel.dart [hunk 3]
```diff
@@ -57,54 +70,202 @@ class _NewChannelState extends State<NewChannel> {
...
+                              style: AppTextStyles.body2Medium,
+                            ),
+                            SizedBox(
+                              height: 20,
+                              child: Switch(
+                                inactiveTrackColor: Color(0XFFEBEBEB),
+                                activeColor: AppColors.zuriPrimaryColor,
+                                value: _value,
+                                onChanged: (val) {
+                                  setState(() {
```
Comment (isaacadariku) [inline lib/ui/view/channel/new_channel/new_channel.dart:222]: Here too

File: lib/ui/view/channel/new_channel/new_channel.dart [hunk 4]
```diff
@@ -57,54 +70,202 @@ class _NewChannelState extends State<NewChannel> {
...
                 ]),
-              ),
-            ));
+          ),
+        ),
+      ),
+    );
+  }
+
+  void validateInput(String input) {
```
Comment (isaacadariku) [inline lib/ui/view/channel/new_channel/new_channel.dart:249]: All validation should be in a file

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Changes requested

File: lib/ui/view/channel/new_channel/new_channel.dart (hunk 4 above)
Comment (usmanabdulmajid) [reply lib/ui/view/channel/new_channel/new_channel.dart]: @isaacadariku  the requested changes has been made

File: lib/ui/view/channel/new_channel/new_channel_viewmodel.dart [hunk 5]
```diff
@@ -1,6 +1,42 @@
+import 'package:flutter/material.dart';
```
Comment (isaacadariku) [inline lib/ui/view/channel/new_channel/new_channel_viewmodel.dart:1]: Nothing should introduce material or cupertino import in the viewmodel, Kindly make use of FormViewModel and read about it, to use the textEditingController there.

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Read on formViewModel and make the changes to it, remove anything that will bring material import to the VM

File: lib/ui/view/channel/new_channel/new_channel_viewmodel.dart (hunk 5 above)
Comment (usmanabdulmajid) [reply lib/ui/view/channel/new_channel/new_channel_viewmodel.dart]: @isaacadariku the requested changes have been made

**Review by isaacadariku — APPROVED** [review]
LGTM

**Review by Blazebrain — APPROVED** [review]
Good to go

Nice improvement boss🤗

**Pull request merged**